### ISSUE TITLE
mvt-ios python error

### ISSUE THREAD

**Issue by larromba** [mvt-project/mvt#336]
Hello,

I ran the following commands:
`idevicebackup2 backup --full backupFolder`
`mvt-ios decrypt-backup backupFolder --destination backupFolderDecrypted`
`mvt-ios download-iocs`
`mvt-ios check-backup backupFolderDecrypted`

At first there was a problem with something being corrupted for the step `Running module Datausage... `, which I fixed with the tips from here: https://github.com/mvt-project/mvt/issues/25

I then ran this command once again:
`mvt-ios check-backup backupFolderDecrypted`

And now receive an error after processing everything from the step: `Running module Datausage... `:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/mvt/common/indicators.py", line 237, in check_domain
    orig_url = URL(url)
               ^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/mvt/common/url.py", line 263, in __init__
    self.domain = self.get_domain()
                  ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/mvt/common/url.py", line 276, in get_domain
    return get_tld(self.url,
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/tld/utils.py", line 510, in get_tld
    domain_parts, non_zero_i, parsed_url = process_url(
                                           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/tld/utils.py", line 309, in process_url
    if fix_protocol and not url.startswith(("//", "http://", "https://")):
                            ^^^^^^^^^^^^^^
AttributeError: 'int' object has no attribute 'startswith'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/mvt-ios", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/mvt/ios/cli.py", line 182, in check_backup
    cmd.run()
  File "/usr/local/lib/python3.11/site-packages/mvt/common/command.py", line 184, in run
    run_module(m)
  File "/usr/local/lib/python3.11/site-packages/mvt/common/module.py", line 184, in run_module
    module.check_indicators()
  File "/usr/local/lib/python3.11/site-packages/mvt/ios/modules/mixed/webkit_resource_load_statistics.py", line 58, in check_indicators
    ioc = self.indicators.check_domain(result["registrable_domain"])
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/mvt/common/indicators.py", line 263, in check_domain
    if ioc["value"].lower() in url:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'int' is not iterable
```

**Comment by Te-k** [comment]
Hi,
That looks like a problem with the `webkit_resource_load_statistics` module, it shouldn't affect the rest of the analysis. I'll work on a fix for it

**Comment by Te-k** [comment]
Hi,
I have improved the module and added some extra checks, so that the module and MVT shouldn't crash anymore. However I think the bug is coming from a change in the database format. Could you share the format of the table `ObservedDomains` and the iOS version? (to do so, get the file path of the observable.db database, then use sqlite3 to open it and get the database schema with the `.schema` command)

Thanks!

**Comment by larromba** [comment]
Thanks for looking into this. Will the fix be in the next release?


**iOS version**

From the `info.plist` it seems this is the iOS version is `16.1.2`:

```
<key>Product Version</key>
  <string>16.1.2</string>
```

**ObservedDomains**

Regarding `observable.db`, I can only see `Manifest.db` in my folder. Is that the correct one? 

When I tried `.schema`, nothing appeared at first:

```
sqlite> .open /Volumes/2\ TB\ Intenso\ External\ USB\ 3.0\ Media/hd/mvt-ios\ iphonebackups/08\:03\:23\ -\ iphone\ SE/Manifest.db 
extra argument: "TB\"
sqlite> .schema
```

I read about the `.schema` command [here](https://sqlite.org/cli.html), and tried:

```
sqlite> .schema main.*
```

which gave me:

```
/* bytecode(addr,opcode,p1,p2,p3,p4,p5,comment,subprog) */;
/* completion(candidate) */;
/* dbstat(name,path,pageno,pagetype,ncell,payload,unused,mx_payload,pgoffset,pgsize) */;
/* fsdir(name,mode,mtime,data) */;
/* fts3tokenize(input,token,start,"end",position) */;
/* generate_series(value) */;
/* json_each("key",value,type,atom,id,parent,fullkey,path) */;
/* json_tree("key",value,type,atom,id,parent,fullkey,path) */;
/* pragma_database_list(seq,name,file) */;
/* pragma_module_list(name) */;
/* sqlite_dbdata(pgno,cell,field,value) */;
/* sqlite_dbpage(pgno,data) */;
/* sqlite_dbptr(pgno,child) */;
/* tables_used(type,schema,name,wr,subprog) */;
```

I also opened `Manifest.db` using the [DB Browser for SQLite](https://sqlitebrowser.org/blog/version-3-12-2-released/), but I'm not sure there's an `ObservedDomains` table:

<img width="471" alt="Screenshot 2023-04-08 at 10 41 43" src="https://user-images.githubusercontent.com/676915/230712363-eb24ab2f-e11f-4913-addb-4ffacf6fc078.png">

Please let me know if you need any further help 👍

**Comment by Te-k** [comment]
I would actually need the schema of a few `observations.db` databases if you have a couple of minutes. Could you share the schema of the `ObservedDomains` for the files `6e/6e9d0cb750a70e7fa10943c134a5f021dab327e7`, `f6/f653134757c17c495b1f2790714a4a41ce366883` and `68/68a018d92ba81499554e4850dea1357e4e8639e9`? (By doing the same `sqlite PATH` and .schema).

Thanks!

**Comment by Te-k** [comment]
Thanks. I am not sure what created the issue but I think the last version should fix it.
Could you try with the latest version from github?
You can install it with `pip3 uninstall mvt` and then `pip3 install git+https://github.com/mvt-project/mvt.git` (feel free to do that in a virtual env if you just want to test it)

**Comment by larromba** [comment]
I can confirm this issue seems to be fixed with version `2.2.3`. Thanks 👍

**Comment by Te-k** [comment]
Thanks !
